src/hooks/useDiscordEvent.ts

```diff
@@ -0,0 +1,50 @@
+import useSWR, { SWRConfiguration } from "swr"
+
+type DiscordEvent = {
+  id: string
+  guildId: string
+  channelId?: string
+  creatorId?: string
+  name: string
+  description?: string
+  scheduledStartTimestamp: string
+  scheduledEndTimestamp?: string
+  privacyLevel: number
+  status: number
+  entityType: number
+  entityId?: string
+  userCount?: number
+  creator?: number
+  entityMetadata?: {
+    location: string
+  }
+  image?: string
+}
+
+const fallbackData = []
+
+const useDiscordEvents = (
```

Comment:
Because of the error logic mentioned above, I want to organize this into a separate hook to be cleaner. However, It could be a good idea to add some comments here to explain the unique error behavior.

Reply:
All right, yes, then let's leave this hook as is (just change `useSWR` to `useSWRImmutable`, please), and add a short comment ;) Also, let's stick to the original SWR return type:

```ts
const swrResponse = useSWRImmutable<DiscordEvent[]>(paltformGuildId ? `/discord/events/${paltformGuildId}` : null)

return {
    ...swrResponse
    data: data.events ?? [],
    // Add comment here...
    error: error || data.error,
}
```